build/pmm-client-docker/entrypoint.py

```diff
@@ -41,15 +44,39 @@ def main():
 
     if PMM_AGENT_SETUP:
         print('Starting pmm-agent setup ...', file=sys.stderr)
-        status = os.system('pmm-agent setup')
-        if status != os.EX_OK:
+        status = subprocess.call(['pmm-agent', 'setup'])
+        print('pmm-agent setup exited with {}.'.format(status), file=sys.stderr)
+        if status != 0:
             sys.exit(status)
 
-    print('Starting pmm-agent ...', file=sys.stderr)
-    sys.stderr.flush()
+    if PMM_AGENT_PRERUN_SCRIPT:
+        print('Starting pmm-agent for prerun script ...', file=sys.stderr)
+        agent = subprocess.Popen(['pmm-agent', 'run'])
+
+        print('Running {} ...'.format(PMM_AGENT_PRERUN_SCRIPT), file=sys.stderr)
+        status = subprocess.call([PMM_AGENT_PRERUN_SCRIPT])
```

Comment:
there are two requirements
1) command like `sh -c "ls;id"` should work fine, just test with `PMM_AGENT_PRERUN_SCRIPT = 'sh -c "ls;id"'`
2) we should see stdout and stderr of command on the main stderr

Reply:
`PMM_AGENT_PRERUN_SCRIPT ` contains file name that should be added via volume or some other means, not the script itself.

stdout and stderr should already be visible. Why you want both stdout and stderr in stderr?

Reply:
> contains file name that should be added via volume
it is not the right thing, please let's just add one call
```suggestion
        status = subprocess.call(shlex.split(PMM_AGENT_PRERUN_SCRIPT))
```

Reply:
```
export PMM_AGENT_PRERUN_SCRIPT='ls -la | grep Dockerfile && echo ls yes || echo ls no; id -u | grep 1000 && echo id yes || echo id no'
./entrypoint.py
```

now produces

```
Running prerun shell script ...
-rw-r--r--  1 aleksi  staff   728 Jun  9 15:48 Dockerfile
ls yes
id no
Prerun shell script exited with 0.
```